pilot/pkg/util/network/ip.go

```diff
@@ -141,9 +141,26 @@ func ResolveAddr(addr string, lookupIPAddr ...lookupIPAddrType) (string, error)
 	return resolvedAddr, nil
 }
 
-// IsIPv6Proxy check the addresses slice and returns true for all addresses are valid IPv6 address
-// for all other cases it returns false
-func IsIPv6Proxy(ipAddrs []string) bool {
+// IsIPv6 checks the addresses slice and returns true if atleast one of the addresses
+// is a valid IPv6 address, for all other cases it returns false.
+func IsIPv6(ipAddrs []string) bool {
```

Comment:
seems we got rid of IsLinkLocalUnicast check?

Reply:
yes. That was reverted yesterday. I merged the same code here

Reply:
the original logic is that return true only if all IP addresses are IPv6, the new logic is that return true if there is at least one IP address is IPv6. And this function is widely used in many places, I think this change may cause some problem.

Reply:
I verified at call sites to see if there would be a problem and at once place only it could and I handled. Can you please let me know if you see any potential issues?

Reply:
In fact I git clone the repo from master branch, and can not correctly install my Istio with `istio-ingressgateway` which has the same situation before the IsIPv6Proxy enhancement was reverted yesterday.
![image](https://user-images.githubusercontent.com/4101246/160844320-209e8e94-e9ca-4fe5-b571-8c568fe43a03.png)

Reply:
Hmm. Means we are missing some test case here. Can you please see if this https://github.com/istio/istio/pull/38177 branch helps you? If not, I will take another look or we can revert this.

Reply:
The same behavior
![image](https://user-images.githubusercontent.com/4101246/160870085-5ff829bc-8d3e-4f1c-8da6-35803a2519e4.png)